packaging/debian/control

```diff
@@ -1,36 +1,39 @@
-Source: google-compute-engine
+Source: gce-compute-image-packages
+Maintainer: Google Cloud Team <gc-team@google.com>
 Section: admin
 Priority: optional
-Maintainer: Google Cloud Team <gc-team@google.com>
 Build-Depends: debhelper (>= 10)
 Standards-Version: 3.9.8
 Homepage: https://github.com/GoogleCloudPlatform/guest-configs
 
 Package: google-compute-engine
 Architecture: all
-Depends: google-compute-engine-oslogin,
+Depends: ${misc:Depends},
+         google-compute-engine-oslogin (>> 20190801),
          google-guest-agent,
          nvme-cli,
          ${misc:Depends}
-Recommends: rsyslog | system-log-daemon
 Provides: irqbalance
-Conflicts: google-compute-engine-jessie,
-           google-compute-engine-init-jessie,
-           google-config-jessie,
-           google-compute-engine-stretch,
-           google-compute-engine-init-stretch,
-           google-config-stretch,
-           google-compute-daemon,
-           google-startup-scripts,
-           irqbalance
```

Comment:
Why remove irqbalance?

Reply:
I'm interested in what's happening here. The conflict with irqbalance has been problematic, but also we've talked about the guest environment handling this better, or possibly having the agent include some IRQ balancing features.

AIUI the conflict exists because we have scripts that do some initial balancing for virtio and gve, but new features are needed (NUMA awareness for one) and also it's run-once, while irqbalance watches for pressure and responds by moving IRQs to other cores.

On the other hand, I understand that irqbalance is fairly naive. There's a lot going on here.

Reply:
@utkarsh2102 how is Ubuntu resolving this conflict? If Ubuntu users expect to be able to use both and that's working for them, that should continue to work. I am not familiar with our IRQ balancing so I can't say how much work it is to modify our balancing to play nice with irqbalance on Debian.

Reply:
> Why remove irqbalance?

Just to be clear, this diff represents that we're dropping the `Conflict` - meaning, we don't think there's one to begin with. Both packages _can_ be installed at the same time.

Now, looking at the history - we never had the Conflict in place since the initial packaging (done back in 2016). We added the `Provides` later in 2018ish, I think. So now when I am submitting the changes here, the diff appears to be dropping the Conflict but it was never there to begin with.

I can't speak for why it was never there to begin with but we can perhaps add it but only if there's a strong rationale to do that. Adding a conflict would mean that a user with guest-configs installed will never be able to install irqbalance on their system - that seems a bit too rough. Do you have a strong rationale as to why this co-installation should _never_ happen?

Reply:
>Do you have a strong rationale as to why this co-installation should never happen?

I don't, I'm not super familiar with this area. I would need to do more research before I can say we should or shouldn't be conflicting with irqbalance.

Reply:
Because there's a script that tries to set the IRQ spread for gve (and virtio-net?) I believe there was a concern that irqbalance would choose to undo what that script does. Depending on which starts first, one or the other may win. There may also be concerns that irqbalance will make less informed decisions than our initial setup script can, though it's currently not taking full advantage of that.

I believe irqbalance is more about continuous balancing as load changes. It's not clear to me if it takes action immediately or only under load.

Ultimately I think it's better if we allow customers to install irqbalance, but I'm not sure how to resolve the real conflict if there is one. Probably we can at least test the behavior to see if irqbalance rebalances from the initial setting, and if it does something wrong if so. And then we can redo the performance testing that depends on these being balanced.